### PR TITLE
Add inputReader for easy reading of edm4hep or lcio files

### PR DESCRIPTION
**Pull request by Zehvogel**
BEGINRELEASENOTES
- Add inputReader for easy reading of edm4hep or lcio files

ENDRELEASENOTES

see discussion in #178

### CODE REVIEW

**Comment by tmadlener** [comment]
Since we are on the topic of making this "easy to use". Is there an easy way to also deal with the conditional attaching of a converter? From a quick look at the code we would need the `algList` and attach it to the first `MarlinWrapper` algorithm? That could also check that the first / only algorithm before that is the `PodioInput` (or do nothing in case it's an `LcioEvent`?

Maybe there is something more clever, but I would be happy with a two step/function approach for the moment.

**Comment by Zehvogel** [comment]
I will add conversion in the other direction in another pr later

File: k4MarlinWrapper/python/k4MarlinWrapper/inputReader.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        if any(not f.endswith(".root") for f in input_files):
+            print("All input files need to have the same format (EDM4hep)")
+            sys.exit(1)
+        read = PodioInput("PodioInput")
+        evtSvc.inputs = input_files
+
+    return read
+
+
+def attach_edm4hep2lcio_conversion(algList, read):
```
Comment (tmadlener) [inline k4MarlinWrapper/python/k4MarlinWrapper/inputReader.py:42]: Can you add a brief description as a docstring that contains the fact that this is pretty much what you want to do if you have used `create_reader` unconditionally?

**Review by tmadlener — COMMENTED** [review]
Apart from a bit of missing documentation I am very happy with this. Thanks :)

File: k4MarlinWrapper/python/k4MarlinWrapper/inputReader.py (hunk 1 above)
Comment (Zehvogel) [reply k4MarlinWrapper/python/k4MarlinWrapper/inputReader.py:41]: done :)

File: k4MarlinWrapper/python/k4MarlinWrapper/inputReader.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    """Attach the edm4hep to lcio conversion if necessary e.g. when using create_reader. Should only be run after algList is complete."""
+    if not isinstance(read, PodioInput):
+        # nothing to convert :)
+        return
+
+     # find first wrapper
+    for alg in algList:
+        if (isinstance(alg, MarlinProcessorWrapper)):
```
Comment (jmcarcell) [inline k4MarlinWrapper/python/k4MarlinWrapper/inputReader.py:49]:
```suggestion
        if isinstance(alg, MarlinProcessorWrapper):
```

Comment (Zehvogel) [reply k4MarlinWrapper/python/k4MarlinWrapper/inputReader.py]: thx

**Review by tmadlener — APPROVED** [review]

**Pull request merged**